libraries/AC_AttitudeControl/AC_PosControl.h

```diff
@@ -29,265 +27,279 @@
+    ///     The time constant also defines the time taken to achieve the maximum acceleration.
+    ///     The function alters the input velocity to be the velocity that the system could reach zero acceleration in the minimum time.
+    void input_pos_vel_accel_xyz(Vector3f& pos);
 
-    /// get_max_speed_up - accessor for current maximum up speed in cm/s
-    float get_max_speed_up() const { return _speed_up_cms; }
-
-    /// get_max_speed_down - accessors for current maximum down speed in cm/s.  Will be a negative number
-    float get_max_speed_down() const { return _speed_down_cms; }
-
-    /// get_vel_target_z - returns current vertical speed in cm/s
-    float get_vel_target_z() const { return _vel_target.z; }
-
-    /// set_max_accel_z - set the maximum vertical acceleration in cm/s/s
-    ///     leash length will be recalculated
-    void set_max_accel_z(float accel_cmss);
-
-    /// get_max_accel_z - returns current maximum vertical acceleration in cm/s/s
-    float get_max_accel_z() const { return _accel_z_cms; }
-
-    /// calc_leash_length - calculates the vertical leash lengths from maximum speed, acceleration
-    ///     called by update_z_controller if z-axis speed or accelerations are changed
-    void calc_leash_length_z();
-
-    /// set_alt_target - set altitude target in cm above home
-    void set_alt_target(float alt_cm) { _pos_target.z = alt_cm; }
+    ///
+    /// Lateral position controller
+    ///
 
-    /// set_alt_target_with_slew - adjusts target towards a final altitude target
-    ///     should be called continuously (with dt set to be the expected time between calls)
-    ///     actual position target will be moved no faster than the speed_down and speed_up
-    ///     target will also be stopped if the motors hit their limits or leash length is exceeded
-    void set_alt_target_with_slew(float alt_cm, float dt);
+    /// set_max_speed_accel_xy - set the maximum horizontal speed in cm/s and acceleration in cm/s/s
+    /// If set accel_limit_cmss limits the maximum correction from the position controller to be less than the maximum lean angle
+    void set_max_speed_accel_xy(float speed_cms, float accel_cmss, float accel_limit_cmss = 0.0f);
 
-    /// set_alt_target_from_climb_rate - adjusts target up or down using a climb rate in cm/s
-    ///     should be called continuously (with dt set to be the expected time between calls)
-    ///     actual position target will be moved no faster than the speed_down and speed_up
-    ///     target will also be stopped if the motors hit their limits or leash length is exceeded
-    ///     set force_descend to true during landing to allow target to move low enough to slow the motors
-    virtual void set_alt_target_from_climb_rate(float climb_rate_cms, float dt, bool force_descend);
+    void set_shaper_tc_xy(float tc_xy_s) {_tc_xy_s = MAX(_tc_xy_s, tc_xy_s);}
```

Comment:
looks like set_shaper_tc_xy() is never called so it should probably be removed or at least get a comment if we know it will be used in the near future.